src/protocol.rs

```diff
@@ -190,6 +190,62 @@ pub fn read_info_field(device: &HidDevice, field: InfoField) -> Result<String, E
         .to_owned())
 }
 
+pub fn run_tap_commands(device: &HidDevice) -> Result<(), Error> {
+    const TAP_REPORT_ID: u8 = 2;
+    const TAP_REPORT_LEN: usize = 2048;
+
+    loop {
+        print!("> ");
+        std::io::stdout().flush().unwrap();
+        let mut tap_command = String::new();
+        std::io::stdin()
+            .read_line(&mut tap_command)
+            .expect("Failed to read line");
+
+        tap_command = tap_command.trim().to_string();
+        if tap_command.len() == 0 {
+            continue;
+        } else if tap_command == "." {
+            break;
+        }
+        let tap_bytes = tap_command.as_bytes();
+
+        // 1 byte report ID + 2 bytes field ID + 1 byte NUL
```

Comment:
Everything from here on is identical to `read_info_field()`, right? Can you make a new function called `run_tap_command()` containing the common code and then call that from both this function and from `read_info_field()`? That'll save a lot of code duplication, and it'll make `read_info_field()` pretty trivial (just a single enum mapping before the function call).